cmd/osm-controller/validate.go

```diff
@@ -46,5 +31,45 @@ func validateCLIParams() error {
 	if webhookName == "" {
 		return errors.Errorf("Invalid --webhook-name value: '%s'", webhookName)
 	}
+
+	return nil
+}
+
+func validateCertificateManagerOptions() error {
+	switch *osmCertificateManagerKind {
```

Comment:
If I was to borrow a primitive from FP it would have been nice to have a map of validators here and pick the right validator based on `osmCertificateManagerKind`

Reply:
ah yea definitely. can try to tackle that as a followup.

Reply:
@michelleN I was just expressing a perspective (I happen to love FP, though it is tough to apply to Go) - just sharing ideas - not suggesting we change this!